web/api/v1/api.go

```diff
@@ -1427,8 +1440,24 @@ func (api *API) rules(r *http.Request) apiFuncResult {
 		return invalidParamError(err, "exclude_alerts")
 	}
 
+	paginationRequest, parseErr := parseListRulesPaginationRequest(r)
+	if parseErr != nil {
+		return invalidParamError(parseErr.err, parseErr.parameter)
+	}
+
 	rgs := make([]*RuleGroup, 0, len(ruleGroups))
+
+	foundToken := false
```

Comment:
Do I understand this right, that if token is set, but the group was removed in between, we will return no rules? Should we fail request instead?

Reply:
Good catch. I will handle that case with a error.